book/python/using-connectors.md

```diff
@@ -0,0 +1,191 @@
+
+The application module is not quite enough for the script to know what settings to use when connecting to wallaroo. Since your application may have more than one connector, the `--conncetor` argument is required for disambiguating which connector in your application setup this script is running as. Once you pass this in, we can autodetect the most common settings.
+
+Many connector scripts will have their own additional arguments, some may be required and others may be optional. In this case we have two, the host and port. These argument names prefix the connector name to ensure that conflicting argument names do not cause problems (should be able to pass all your wallaroo arguments in without conflict for example).
+
+To illustrate this, let's look at what the `udp_source` command line parameters might look like:
+
+```bash
+./wallaroo/connectors/udp_source --application-module celsius --connector celsius_feed --celsius_feed-host 127.0.0.1 --celsius_feed-port 6789
+```
+
+In this example, we are running a UDP source for our application whic listens to a UDP socket on 127.0.0.1 port 6789 and forwards the data in each packet to wallaroo as a message.
+
+Since the included connector scripts are self-contained, feel free to copy it over to your application tree to use them directly and make path management a little easier.
+
+## Starting With Prebuilt Connectors
+
+We have a number of [prebuild connectors](https://github.com/WallarooLabs/wallaroo/tree/{{book.wallaroo_version}}/connectors/). Here is a short description of each one.
+
+### Kafka
+
+This connector provides Kafka source and sink support. This uses the [kafka library](https://pypi.org/project/kafka/). Consumer group based sources are supported to allow easier management of your source's progress through a topic.
+
+### Kinesis
+
+AWS Kinesis is supported via the [boto3 library](https://pypi.org/project/boto3/). This script will expect AWS credentials to be setup as described in their documentation.
+
+### Redis
+
+Redis has many ways to be used as a source and a sink. We've provided two starting points to show how a source and sink can work. The script is very easy to modify and uses the well maintained [redis library](https://pypi.org/project/redis/).
+
+For the source, we show an example using the subscription support offered by Redis pubsub and it's a natural fit for many stream processing applications. Check out `redis_subscriber_source` for [details](https://github.com/WallarooLabs/wallaroo/tree/{{book.wallaroo_version}}/connectors/redis_subscriber_source).
+
+For the sink, we show that not everything needs to look exactly like a stream to be used with Wallaroo. We use a give hash in redis and set key, value pairs on that hash for each tuple passed to the sink. This is very handy for those cases that want to keep track of the latest value. See `redis_hash_sink` for [details](https://github.com/WallarooLabs/wallaroo/tree/{{book.wallaroo_version}}/connectors/redis_hash_sink).
```

Comment:
"given hash"
"(key, value)"

Reply:
Maybe I'll use key-value here since it's not a syntactical thing at this point?

Reply:
I'm also thinking "on the target hash" might be better.

Reply:
sounds good